features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java

```diff
@@ -124,16 +122,19 @@ public ElasticFlowRepository(MetricRegistry metricRegistry, JestClient jestClien
     }
 
     @Override
-    public void persistNetFlow5Packets(Collection<? extends NF5Packet> packets, FlowSource source) throws FlowException {
-        LOG.debug("Converting {} Netflow 5 packets from {} to flow documents.", packets.size(), source);
-        final List<FlowDocument> flowDocuments = packets.stream()
+    public <P> void persist(final Collection<? extends P> packets, final FlowSource source, final Converter<P> converter) throws FlowException {
+        LOG.debug("Converting {} flow packets from {} to flow documents.", packets.size(), source);
```

Comment:
Can we make sure the calls to `convert` are timed? Either within the `logEnrichementTimer` or a new timer.